lib/reader.rb

```diff
@@ -0,0 +1,33 @@
+require 'json'
+
+class Reader
+
+  def initialize(path, file = nil)
+    @path = path
+    @file = file
+  end
+
+  def read_contacts
+    file.nil? ? parse(read) : parse(tests_read)
+  end
+
+  private
+
+  attr_reader :path, :file
+
+  def parse(contacts)
+    contacts == '' ? [] : JSON.parse(contacts)
+  end
+
+  def read
+    file = File.open(path, "r")
```

Comment:
That's what Skim mentioned before going home, gotta try that out.
(blocks and such) :+1: